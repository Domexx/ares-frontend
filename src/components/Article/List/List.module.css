.List {
  grid-column: 1 / -1;
  gap: 16px;
}
