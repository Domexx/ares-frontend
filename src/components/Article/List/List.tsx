import React from 'react';
import s from './List.module.css';

import { ArticleGridItem } from '../../../shared/interfaces/ArticleGridItem';
import ListItem from '../ListItem';

export type Props = {
  items: ArticleGridItem[];
};

const List: React.FC<Props> = ({ items }) => {
  const listItems = items.map((item: ArticleGridItem, index: number) => {
    return (
      <ListItem
        title={item.title}
        description={item.description}
        image={item.image}
        date={item.date}
        category={item.category}
        content={item.content}
        // eslint-disable-next-line react/no-array-index-key
        key={index}
      />
    );
  });

  return <div className={`${s['List']} u-grid `}>{listItems}</div>;
};

export default List;
