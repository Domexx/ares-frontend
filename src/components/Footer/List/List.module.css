/* === === === === Footer/List === === === === */

.List {
  margin-top: 0;
  padding: 0;
}

.List:nth-of-type(1) {
  grid-column: 5 / span 2;
}

.List:nth-of-type(2) {
  grid-column: 7 / span 2;
}

.List-headline {
  font-size: 14px;
}

@media (--laptop) {
  .List:nth-of-type(1) {
    grid-column: 6 / span 2;
  }

  .List:nth-of-type(2) {
    grid-column: 9 / span 2;
  }
}

@media (--tablet-landscape) {
  .List:nth-of-type(1) {
    grid-column: 9 / span 3;
  }

  .List:nth-of-type(2) {
    grid-column: 13 / span 3;
  }
}

@media (--phone) {
  .List {
    margin-top: 15px;
    padding-top: 15px;
    border-top: 1px solid var(--clr-dark-600);
  }

  .List:nth-of-type(1) {
    grid-column: 2 / span 7;
  }

  .List:nth-of-type(2) {
    grid-column: 9 / span 7;
  }
}
