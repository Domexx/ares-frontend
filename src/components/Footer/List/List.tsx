/* eslint-disable react/no-array-index-key */
import React from 'react';
import s from './List.module.css';

import { FooterListChildren, FooterListItem } from '../../../shared/interfaces/FooterItem';
import ListItem from '../ListItem';

export type Props = {
  items: FooterListItem[];
};

const List: React.FC<Props> = ({ items }) => {
  const listItems = items.map((item: FooterListItem, index: number) => {
    return (
      <ul key={index} className={s['List']}>
        <div key={index} className={s['List-headline']}>
          {item.headline}
        </div>

        {item.children.map((children: FooterListChildren) => {
          return <ListItem name={children.name} url={children.link} key={index} />;
        })}
      </ul>
    );
  });

  // eslint-disable-next-line react/jsx-no-useless-fragment
  return <>{listItems}</>;
};

export default List;
