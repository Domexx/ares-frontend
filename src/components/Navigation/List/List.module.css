/* === === === === Navigation/List === === === === */

.List {
  margin-top: 0;
}
