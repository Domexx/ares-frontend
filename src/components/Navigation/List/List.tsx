import React from 'react';
import s from './List.module.css';

// eslint-disable-next-line @typescript-eslint/ban-types
export type Props = {};

const List: React.FC<Props> = () => {
  return <div className={s['list']}>List Works!</div>;
};

export default List;
