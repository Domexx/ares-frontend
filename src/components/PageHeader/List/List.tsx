import React from 'react';
import clsx from 'clsx';
import s from './List.module.css';

import { PageHeaderItem } from '../../../shared/interfaces/PageHeaderItem';
import ListItem from '../ListItem';

export type Props = {
  main: PageHeaderItem[];
  secondary: PageHeaderItem[];
  className?: string;
};

const List: React.FC<Props> = ({ main, secondary, className }) => {
  const classes = clsx(className);
  const mainListItems = main.map((item: PageHeaderItem, index: number) => {
    return (
      // eslint-disable-next-line react/no-array-index-key
      <ListItem className={classes} name={item.name} icon={item.icon} url={item.url} key={index} />
    );
  });

  const secondaryListItems = secondary.map((item: PageHeaderItem, index: number) => {
    return (
      // eslint-disable-next-line react/no-array-index-key
      <ListItem className={classes} name={item.name} icon={item.icon} url={item.url} key={index} />
    );
  });

  return (
    <div className={`${s['List']} u-grid`}>
      <ul className={s['List-main']}>{mainListItems}</ul>
      <ul className={s['List-secondary']}>{secondaryListItems}</ul>
    </div>
  );
};

export default List;
