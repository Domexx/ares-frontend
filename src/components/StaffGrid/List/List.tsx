/* eslint-disable @typescript-eslint/no-shadow */
/* eslint-disable react/no-array-index-key */

import React from 'react';
import s from './List.module.css';

import ListItem from '../ListItem';
import { Member, StaffGridItem } from '../../../shared/interfaces/StaffGridItem';

export type Props = {
  items: StaffGridItem[];
};

const List: React.FC<Props> = ({ items }) => {
  const listItems = items.map((item: StaffGridItem, index: number) => {
    return (
      <div key={index} className={s['List']}>
        <div className={s['List-headline']}>{item.title}</div>
        <div className={s['List-content']}>
          {item.members.map((member: Member, index: number) => {
            return (
              <ListItem
                name={member.name}
                key={index}
                look={member.look}
                motto={member.motto}
                online={member.online}
                rank={member.rank}
              />
            );
          })}
        </div>
      </div>
    );
  });

  // eslint-disable-next-line react/jsx-no-useless-fragment
  return <>{listItems}</>;
};

export default List;
