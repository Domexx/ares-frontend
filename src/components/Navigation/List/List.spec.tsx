import React from 'react';

import { render } from '@testing-library/react';

import List from './List';

describe('List Component', () => {
  it('should render properly', () => {
    const { container } = render(<List />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
