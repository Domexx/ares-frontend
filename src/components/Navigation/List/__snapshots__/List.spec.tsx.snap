// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component should render properly 1`] = `
<div
  class="list"
>
  List Works!
</div>
`;
